network/proto/senders.go

```diff
@@ -63,10 +70,37 @@ func (s defaultMessageSender) sendTransactionListQuery(peer p2p.PeerID, blockDat
 }
 
 func (s defaultMessageSender) sendTransactionList(peer p2p.PeerID, transactions []dag.Transaction, blockDate time.Time) {
-	envelope := createEnvelope()
+	if len(transactions) == 0 {
+		envelope := createEnvelope()
+		envelope.Message = &transport.NetworkMessage_TransactionList{TransactionList: &transport.TransactionList{Transactions: toNetworkTransactions([]dag.Transaction{}), BlockDate: uint32(blockDate.Unix())}}
+		s.doSend(peer, &envelope)
+		return
+	}
+	// When the DAG grows it the transactions might not fit in 1 network message (defined by p2p.MaxMessageSizeInBytes),
+	// so we have to split it up in pages. Protobuf can calculate the size of a message, so we can check whether it will fit in the max. message size.
+	// However, this can be quite expensive since it will marshal the message to the wire format (thus, marshalling will happen twice).
+	// Furthermore, we'd have to do this for every transaction we add to the message to find the max. number of TXs that
+	// will fit in the message.
+	// Since this sounds all horribly inefficient, we just calculate the size of a network message with 1 transaction,
+	// and use that to guesstimate the number of transactions that will fit in a message, decreased by a safety margin.
+	// Since transactions just contain metadata of the transaction (and not the payload itself), all transactions should
+	// serialize to more or less the same number of bytes, making the calculation safe.
+	sizeMsg := createEnvelope()
```

Comment:
this calculation can be done in the initializer.

Things sounds horribly inefficient, but given current CPU power, how inefficient is it compared to other operations (eg compared to cryptographic operations)?